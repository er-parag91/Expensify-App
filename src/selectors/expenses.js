import moment from 'moment';
//Get visible expenses at start

export default (expenses, {
    text,
    sortBy,
    startDate,
    endDate
}) => {

    return expenses.filter(expense => {
        const createdAtMoment = moment(expense.createdAt)
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment) : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment) : true;
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1
        } else if (sortBy === 'amount') {
            return a.amount > b.amount ? -1 : 1
        }
    })
}