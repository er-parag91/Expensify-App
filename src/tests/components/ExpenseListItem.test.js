import ExpenseListItem from '../../components/ExpenseListItem';
import { shallow } from 'enzyme';
import React from 'react';
import expenses from '../fixtures/fixtures';
import toJSON from 'enzyme-to-json';

test('should render a ExpenseListItem correctly', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
})