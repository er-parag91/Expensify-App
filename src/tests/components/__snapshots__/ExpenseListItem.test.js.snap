// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a ExpenseListItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h1>
      Gum
    </h1>
  </Link>
  <p>
    195
     : 
  </p>
</div>
`;
